pkg/common/cns-lib/volume/manager.go

```diff
@@ -1275,6 +1291,20 @@ func (m *defaultManager) deleteVolumeWithImprovedIdempotency(ctx context.Context
 	volumeOperationRes := taskResult.GetCnsVolumeOperationResult()
 	if volumeOperationRes.Fault != nil {
 		faultType = ExtractFaultTypeFromVolumeResponseResult(ctx, volumeOperationRes)
+		faultTypeErrSplitString := strings.Split(faultType, ".")
+		faultTypeErr := faultTypeErrSplitString[len(faultTypeErrSplitString)-1]
+		// If volume is not found on host, but is present in CNS DB, we will get NotFound fault.
+		// In such a case, send back success as the volume is already deleted.
+		notFoundErrSplitString := strings.Split(((reflect.TypeOf(vim25types.NotFound{})).String()), ".")
+		// vim25types gives value of format *type.XXX.
+		// In this case it will be *type.NotFound
+		// Extract Notfound from the vim25types error.
+		notFoundErr := notFoundErrSplitString[len(notFoundErrSplitString)-1]
+		if faultTypeErr == notFoundErr {
```

Comment:
why are we overcomplicating when in the end we just want to compare with "NotFound"?
Can you simplify this?

Reply:
IMO it will be better to derive the "NotFound error" from govmomi's `vim25types`. This is to ensure that if it changes in future, we are still referring to the correct return value. If we simply hard code "NotFound" string in our code, it may cause errors in future if it changes on CNS side.

Reply:
these faults are kind of APIs, a lot of internal and external logic depends on this fault. Do not worry about getting it changed.